Jenyus-Org/graphql-utils
# [ENHANCEMENTS] resolveFieldsMap Helper, Wildcard Selector & Updated Args Positions

# Overview

## General

 - Move `FieldSelection` interface to `helpers.ts`.

## Updates

 - Add `atRoot` argument to `hasFields` to check for field only at root-level.
 - Use `resolveFields` in `hasFields` to avoid duplicate logic.
 - Use new helper `resolveFieldMap` and `fieldMapToDot` (see below) in `resolveFields` to avoid duplicate logic.

## New Helpers

### `resolveFieldMap`

Resolves all the fields found in the given `GraphQLResolveInfo` object, and returns a nested objects with the located fields. This new helper is used as the basis for most other operations in `graphql-utils`, and also allows to be passed arguments `parent` and `deep` for more fine-grain control over the results.

### `fieldMapToDot`

Remaps nested objects such as those returned by `resolveFieldMap` to a dot-notated array, as those returned by `resolveSelections` and `resolveFields`.

## Tests

 - Add test for `hasFields` to check non-existent fields.
 - Add test for `resolveSelections` to check non-existent fields.
 - Add test for wildcards with `*` and `**` (deep) selections in `resolveSelections`.
